SimTracker/TrackerHitAssociation/src/TrackerHitAssociator.cc

```diff
@@ -114,7 +114,7 @@ void TrackerHitAssociator::makeMaps(const edm::Event& theEvent, const TrackerHit
           theEvent.labelsForToken(cfToken, labels);
           std::string trackerContainer(labels.productInstance);
```

Comment:
@ngrenz The value of `tofBin` variable depends only from `labels`, which depends only from `cfToken`. Thefore you can set the `tofBin` variable (including the `strstr()` call) once for each `cfToken`, outside the loop over `thisContainerHits` (i.e. between lines 104-107).

Now, after getting rid of (most) dynamic memory allocations, you might also want to check if moving the last `if` outside the loop of `thisContainerHits` has any effect, i.e.
```
for(<A>) {
  if(<B>) {
    <C>
  } else {
    <D>
  }
}

====>

if(<B>) {
  for(<A>) {
    <C>
  }
} else {
  for(<A>) {
    <D>
  }
}
```
It could be that the CPU branch predictor does a decent-enough job compared to the cost of finding/inserting an element from/to a map (+vector push_back) yielding negligible improvement in time. Actually these costs could already hide the cost of the repeated C-string comparisons (i.e. my first suggestion above). But I think it's nevertheless worthwhile to test them.